src/pgrepr/src/value.rs

```diff
@@ -297,6 +324,10 @@ impl Value {
                 }
                 Ok(postgres_types::IsNull::No)
             }
+            // TODO: what is the binary format of vector types?
+            Value::Int2Vector { .. } => {
+                Err("binary encoding of int2vector is not implemented".into())
```

Comment:
tbd! I took this PR over from @mkysel, which was a request from @andrioni; I'm not totally sure what the exact usecase/driver is, but certainly something I am glad to do as a fast-follow.

Reply:
some PG system tables use this format. See #10024

Reply:
We need it for Superset support, which means psycopg2 (via SQLAlchemy), which always (AFAIK) uses the text protocol. The query they run actually depends on the precise weird behavior of `int2vector`: it behaves as as `int2[]` for SQL operations, but it gets sent as a space-delimited string in the wire.

Sequelize (NodeJS ORM) also depends on this behavior, but AFAIK it also only uses text.

Reply:
Might be worthwhile to open an issue if it won't be handled in this PR, unless you have already.

Reply:
I'm reluctant to open an issue because it makes it seem like "work we want to do," when it's moreso "work we only want to do if we must." it's unclear if there's any benefit at all given that it sounds like most drivers use text encoding.